### ISSUE TITLE
Tool to detect API/ABI non-compliant changes, and integrate with CI.

### ISSUE THREAD

**Issue by lalitb** [open-telemetry/opentelemetry-cpp#1996]
As discussed in [today's Feb 22, 2023](https://docs.google.com/document/d/1i1E4-_y4uJ083lCutKGDhkpi3n4_e774SBLi9hPLocw/edit#heading=h.daa6czvwmr28) community meeting:

- Need tooling to detect ABI changes in CI, instead of relying on code review only
- There was some research done earlier on one of the promising - [ABI Compliance Checker (lvc.github.io)](https://lvc.github.io/abi-compliance-checker/), but good to compare other alternatives before finalizes.

**Comment by ThomsonTan** [comment]
Does this mean it is always recommended for the user to build with the same version of API and SDK?
> otel-cpp doesn't guarantee ABI compatibility for SDK code

**Comment by lalitb** [comment]
> Does this mean it is always recommended for the user to build with the same version of API and SDK?

No. This actually means the SDK specific component ( e.g. Processor and Exporter Interfaces) can be modified in ABI breaking way. So people having external exporters or processors are expected to rebuild them to be compatible with new SDK version.

**Comment by marcalff** [comment]
Tooling added in a dedicated repository:

* https://github.com/marcalff/opentelemetry-cpp-abi

**Comment by owent** [comment]
Could we publish these reports to somewhere?

**Comment by marcalff** [comment]
> Could we publish these reports to somewhere?

Not yet, but will look into this.
